### PR TITLE
Fix transaction paging

### LINKED ISSUE #2332: Paging of transactions is broken
## Steps to reproduce

1. Have more than 20 transactions in the transactions tab
2. Scroll down

## Expected behaviour

More transactions should load (paging)

## Actual behaviour

Nothing really happens

### PR DESCRIPTION
**Pull request by chmanie**
## Description

This PR fixes the transaction list paging. It's based on #2127 and should probably only be reviewed after that one is merged.

Here we fix the loading by actually passing in the correct element into Apollo's pagination loader. As the old solution was quite cumbersome and brittle, I completely rewrote the caching to facilitate Apollo functions instead of custom written ones. This included adding a cache query-field merge function.
Also, I am now using a more react-y approach for setting the root element for the intersection observer. It's a little bit better as the code is in the same place as the actual element, but for a proper solution something like [`react-intersection-observer`](https://www.npmjs.com/package/react-intersection-observer) should be used.

To actually test this properly (also for the reviewers), I added the creation of transaction entries in the database for the `create-data` script.

## Testing

Re-create your dev environment after checking out this branch.

Go to the UserHub, click Transactions and scroll all the way down.

## Diffs

**New stuff** ✨

- Transactions are now created in the `create-data` script.

**Changes** 🏗

* Page loading was fixed and the state/caching strategy for the pagination was completely rewritten. Now there's a loading spinner showing at the bottom of the transactions, showing when there's more to load.

**Deletions** ⚰️

* Weird insersection observer code

Closes #2332

### CODE REVIEW

File: scripts/create-data.js [hunk 1]
```diff
@@ -369,15 +424,42 @@ const mintTokens = async (
 
   // mint
   const mintTokens = await colonyClient.mintTokens(amount);
-  delay();
+  await delay();
```
Comment (chmanie) [inline scripts/create-data.js:427]: `delay`s without `await` don't do anything

**Comment by chmanie** [comment]
I found an issue with this and it seems it has to be done properly. Putting back in draft state

**Comment by chmanie** [comment]
This is considered done now and can be reviewed, once #2127 is merged.

**Review by iamsamgibbs — COMMENTED** [review]
This is working great!

https://github.com/user-attachments/assets/11cf5ea3-c03b-496a-b1fd-df3ae4076737

I made some changes to `create-data.js` on master recently so looks like it needs updating again here, but otherwise is working nicely!

**Review by mmioana — APPROVED** [review]
Really cool stuff, @chmanie 🎉 

https://github.com/user-attachments/assets/e040a16f-6bbe-42d5-90bb-81f5b0cfdc3f

**Review by iamsamgibbs — APPROVED** [review]
🎉 Still working!

https://github.com/user-attachments/assets/fbc9f348-2992-45fb-a479-843b27bb69de

**Pull request merged**